src/translate.rs

```diff
@@ -21,7 +21,14 @@
 //! ```
 //!
 //! `ToGlibPtr`, `FromGlibPtrNone`, `FromGlibPtrFull` and `FromGlibPtrBorrow` work on `gpointer`s
-//! and support different modes of ownership transfer.
+//! and ensure correct ownership of values on the Rust side
+//! according to [Glib ownership transfer rules](https://gi.readthedocs.io/en/latest/annotations/giannotations.html).
+//!
+//! `FromGlibPtrNone` and `FromGlibPtrFull`
```

Comment:
There's also `from_glib_borrow()` which might be worth mentioning somewhere else here :) Especially with that one it's important that it should only be used from C callbacks, and that the return value must never be passed outside the scope of the callback, and that only immutable references to it must be passed to safe Rust code.

Reply:
> only immutable references to it must be passed to safe Rust code

Where does that come from? To cover all cases where the refcount changes?

Reply:
It would otherwise be unsafe. We only own the reference during the scope of the callback so if Rust code takes the object and stores it away somewhere then it will have a dangling pointer once the callback has returned. For that reason we must only pass a reference away, lend it, to Rust code. The Rust code does not own it but only borrows it.